hack/deploy/configure-values.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script interrogates the AKS cluster and Azure resources to generate 
+# the karpenter-values.yaml file using the karpenter-values-template.yaml file as a template.
+
+# Check the cluster name and resource group are provided
+if [ "$#" -ne 2 ]; then
+    echo "Usage: $0 <cluster-name> <resource-group>"
+    exit 1
+fi
+
+echo "Configuring karpenter-values.yaml for cluster $1 in resource group $2 ..."
+
+AZURE_CLUSTER_NAME=$1
+AZURE_RESOURCE_GROUP=$2
+
+AZURE_LOCATION=$(az aks show --name "$AZURE_CLUSTER_NAME" --resource-group "$AZURE_RESOURCE_GROUP" | jq -r ".location")
```

Comment:
nit: is there a good way to store/reuse the output of:
- `az aks show --name "$AZURE_CLUSTER_NAME" --resource-group "$AZURE_RESOURCE_GROUP"`?

Reply:
Good idea, done